appservice/src/createAppService/AppServicePlanSkuStep.ts

```diff
@@ -16,31 +16,47 @@ type ExtendedSkuDescription = WebSiteManagementModels.SkuDescription & { label?:
 
 export class AppServicePlanSkuStep extends AzureWizardPromptStep<IAppServiceWizardContext> {
     public async prompt(wizardContext: IAppServiceWizardContext): Promise<void> {
-        let skus: ExtendedSkuDescription[] = wizardContext.advancedCreation ? this.getRecommendedSkus().concat(this.getAdvancedSkus()) : this.getRecommendedSkus();
-        if (wizardContext.newSiteKind === AppKind.functionapp) {
-            skus.push(...this.getElasticPremiumSkus());
-        } else if (wizardContext.newSiteKind?.includes(AppKind.workflowapp)) {
-            skus = this.getWorkflowStandardSkus();
-        }
-
+        let recommendedSkus = this.getRecommendedSkus();
         const regExp: RegExp | undefined = wizardContext.planSkuFamilyFilter;
         if (regExp) {
-            skus = skus.filter(s => !s.family || regExp.test(s.family));
+            recommendedSkus = recommendedSkus.filter(s => !s.family || regExp.test(s.family));
```

Comment:
You've duplicated this family filter logic here and several lines down - just do it once for all skus

Reply:
In order to set different group names for recommended and advanced sku list, I need to add both sku lists to `pricingTiers` separately. Combining the skus early on will not allow me to do this.